initfiles/bin/check_executes

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+usage() {
+  echo "Usage: check-executes [options] -- cmd args"
+  echo "    -d <directory>     Mounted directory to store post-mortem info in"
+  echo "    -f <file>          Specifies a file to preserve on post-mortem"
+}
+
+PMD_DIRECTORYBASE=${HPCC_DEBUG_PATH}
+[[ -z ${PMD_DIRECTORYBASE} ]] && PMD_DIRECTORYBASE=$(pwd)
+PMD_PROGNAME=
+PMD_COPYFILES=()
+PMD_DALISERVER=
+PMD_WORKUNIT=
+
+while [ "$#" -gt 0 ]; do
+  arg=$1
+  if [[ ${arg:0:1} == '-' ]]; then
+    case "${arg:1:1}" in
+      -) shift
+         PROGNAME=$1
+         shift
+         break
+         ;;
+      d) shift;
+         PMD_DIRECTORYBASE=$1
+         ;;
+      f) shift;
+         PMD_COPYFILES+=($1)
+         ;;
+      *) usage
+         exit
+         ;;
+    esac
+  else
+    usage
+    exit
+  fi
+  shift
+done
+
+if [[ -z $PMD_PROGNAME} ]] ; then
+  usage
+  exit
+fi
+
+# Scan managed process parameters for additional information
+for (( arg=1; arg <= "$#"; arg++ )); do
+  optname=${!arg%=*}
+  optval=${!arg#*=}
+  if [[ ${optname} == '--config' ]]; then
+    PMD_COPYFILES+=(${optval})
+  elif [[] ${optname} == '--daliServer' ]]; then
+    PMD_DALISERVER=${optval}
+  elif [[] ${optname} == '--workunit' ]]; then
+    PMD_WORKUNIT=${optval}
+  fi
+done
+
+ulimit -c unlimited
+
+# Execute the main program, defaulting postmortem logging on (can be overriden by program's config file)
+${PROGNAME} --logging.postMortem=1000 "$@" 
+
+# If it did not exit cleanly, copy some post-mortem info
+retVal=$?
+if [ $retVal -ne 0 ]; then
+  POST_MORTEM_DIR=${PMD_DIRECTORYBASE}/$(hostname)/$(date -Iseconds)
+  mkdir -p ${POST_MORTEM_DIR}
```

Comment:
this may not be unique enough where multiple processes all fail within same second (e.g. some/all thor workers can do).
Could use mktemp -d to ensure unique.

Reply:
They'd have to all be running on the same pod